src/translib/intf_app.go

```diff
@@ -423,6 +424,49 @@ func (app *IntfApp) getSpecificAttr(targetUriPath string, ifKey string, oc_val *
 		} else {
 			return true, e
 		}
+        case "/openconfig-interfaces:interfaces/interface/state/admin-status":
+            val, e := app.getIntfAttr(ifKey, PORT_ADMIN_STATUS)
+                if len(val) > 0 {
+                        switch val {
+                        case "up":
+                                oc_val.AdminStatus = ocbinds.OpenconfigInterfaces_Interfaces_Interface_State_AdminStatus_UP
+                        case "down":
+                                oc_val.AdminStatus = ocbinds.OpenconfigInterfaces_Interfaces_Interface_State_AdminStatus_DOWN
+                        default:
+                                oc_val.AdminStatus = ocbinds.OpenconfigInterfaces_Interfaces_Interface_State_AdminStatus_UNSET
+                        }
+                        return true, nil
+                } else {
+                        return true, e
+                }
+        case "/openconfig-interfaces:interfaces/interface/state/mtu":
+            val, e := app.getIntfAttr(ifKey, PORT_MTU)
+                if len(val) > 0 {
+                    v, e := strconv.ParseUint(val, 10, 16)
+                    if  e == nil {
+                        oc_val.Mtu = (*uint16)(unsafe.Pointer(&v))
+                        return true, nil
+                    }
+                }
+                return true, e
+        case "/openconfig-interfaces:interfaces/interface/state/ifindex":
+            val, e := app.getIntfAttr(ifKey, PORT_INDEX)
+                if len(val) > 0 {
+                    v, e := strconv.ParseUint(val, 10, 32)
+                    if  e == nil {
+                        oc_val.Ifindex = (*uint32)(unsafe.Pointer(&v))
```

Comment:
Just a suggestion. Can we pass the address of variable v to ifIndex, and check it? Because it's safe in GO to pass the address of local variable, Internally it will do escape analysis and moves it to heap, if it's a local variable.

Reply:
Yes. That's what is happening here. The address of local variable "v" is assigned to "oc_val.Ifindex" which is a uint32 pointer.

Reply:
Is unsafe.Pointer() is mandatory, Would it work if we do oc_val.Ifindex = &v?

Reply:
This is resolved.

Reply:
No, it didn't work, because "v" is of type (uint64 *) and we can't typecast it to (uint32 *) directly. So, unsafe.Pointer() makes it a (void *) which we can then typecast to whatever type we want it to be.